### PR TITLE
 intellij module-based add docker support

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/AddDockerSupportAction.java [hunk 1]
```diff
@@ -45,26 +47,33 @@
 import java.io.IOException;
 import java.nio.file.Paths;
 import java.util.List;
+import java.util.Optional;
 
 public class AddDockerSupportAction extends AzureAnAction {
     private static final String NOTIFICATION_GROUP_ID = "Azure Plugin";
     private static final String NOTIFICATION_TITLE = "Add Docker Support";
     private Project project;
```
Comment (Eskibear) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/AddDockerSupportAction.java]: `project` can be removed

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/dockerhost/DockerHostRunSettingsEditor.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public DockerHostRunSettingsEditor(Project project) {
     protected void resetEditorFrom(@NotNull DockerHostRunConfiguration containerLocalRunConfiguration) {
         if (containerLocalRunConfiguration.isFirstTimeCreated()) {
             if (MavenRunTaskUtil.isMavenProject(project)) {
-                MavenRunTaskUtil.addMavenPackageBeforeRunTask(containerLocalRunConfiguration);
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/dockerhost/DockerHostRunSettingsEditor.java]: revert the change of api
